src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -169,6 +188,24 @@ protected ConnectionBufferPool WebSocketBufferPool
             get { return _bufferPool; }
         }
 
+        HashAlgorithm HashAlgorithm
+        {
+            [SecurityCritical]
+            get
+            {
+                if (_hashAlgorithm == null)
+                {
+                    _hashAlgorithm = SHA512.Create();
+                }
+                else
+                {
+                    _hashAlgorithm.Initialize();
+                }
```

Comment:
Could you explain the `else` behaviour here? 
Feels like `_hashAlgorithm.Initialize();` should go into `if (_hashAlgorithm == null)`, but I don't know the usage pattern of HashAlgorithm well enough to make a definitive comment.